### PR TITLE
AWS Lambda support for Java 8 (Corretto) and custom runtimes on Amazon Linux 2

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes https://github.com/terraform-providers/terraform-provider-aws/issues/14657.
Relates: https://github.com/terraform-providers/terraform-provider-aws/issues/14601.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_lambda_function: Add support for `java8.al2` and `provided.al2` runtime values
resource/aws_lambda_layer_version: Add support for `java8.al2` and `provided.al2` runtime values
data-source/aws_lambda_layer_version: Add support for `java8.al2` and `provided.al2` runtime values
```

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS=-run='TestAccAWSLambdaFunction_runtimes'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSLambdaFunction_runtimes -timeout 120m
=== RUN   TestAccAWSLambdaFunction_runtimes
=== PAUSE TestAccAWSLambdaFunction_runtimes
=== CONT  TestAccAWSLambdaFunction_runtimes
--- PASS: TestAccAWSLambdaFunction_runtimes (408.56s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	408.597s
```

Replaces the existing `aws_lambda_function` runtime acceptance tests with a single new test `TestAccAWSLambdaFunction_runtimes` that uses the new AWS SDK `_Values()` functionality for string enumerations.

### CODE REVIEW

File: aws/resource_aws_lambda_function.go [hunk 1]
```diff
@@ -25,29 +25,11 @@ import (
...
-	lambda.RuntimeNodejs12X,
-	lambda.RuntimeProvided,
-	lambda.RuntimePython27,
-	lambda.RuntimePython36,
-	lambda.RuntimePython37,
-	lambda.RuntimePython38,
-	lambda.RuntimeRuby25,
-	lambda.RuntimeRuby27,
-}
+	validation.StringNotInSlice([]string{lambda.RuntimeNodejs}, false),
```
Comment (ewbankkit) [inline aws/resource_aws_lambda_function.go:31]: Retained for backwards compatibility but given that there are now many more runtimes that are at end-of-life we may want to remove this additional check.

Comment (bflad) [reply aws/resource_aws_lambda_function.go:31]: Let's go ahead and remove the "all but" testing of these -- if AWS wants to officially remove support for these, they can properly remove the values from the enumeration type in the API model 👍

Comment (ewbankkit) [reply aws/resource_aws_lambda_function.go:31]: The provider documentation doesn't explicitly list the values for this attribute but instead references https://docs.aws.amazon.com/lambda/latest/dg/API_PublishLayerVersion.html#SSS-PublishLayerVersion-request-CompatibleRuntimes which shows all the values in the API enumeration, so there's nothing to change there.
I will ensure that the examples in the documentation use currently valid values (I saw a reference to `nodejs8.10`).

Comment (ewbankkit) [reply aws/resource_aws_lambda_function.go]: Done.

**Comment by ewbankkit** [comment]
Rebased, made suggested changes are re-ran acceptance test:

```console
$ make testacc TEST=./aws/ TESTARGS=-run='TestAccAWSLambdaFunction_runtimes'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSLambdaFunction_runtimes -timeout 120m
=== RUN   TestAccAWSLambdaFunction_runtimes
=== PAUSE TestAccAWSLambdaFunction_runtimes
=== CONT  TestAccAWSLambdaFunction_runtimes
--- PASS: TestAccAWSLambdaFunction_runtimes (412.94s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	412.981s
```

**Pull request merged**